### DISCUSSION TITLE
i18next value containing Interpolation reference get's changed during AutoTranslation

### DISCUSSION THREAD

**Discussion by alikleit** [WeblateOrg/weblate#8673] [General]
I have a list of components that are of `i18n JSON`  format, most of these components include a reference string like `$t(shared.entities:product.title)`. With `AutoTranslation` enabled, you'd expect for that reference to stay the same for the destination languages. However, for some reason weblate auto translation is changing that string into another reference string that is translated to another keys. For example:

I have the following components:

`menu` that contains key `products` with value `$t(shared.entities:product.title)`
`organization` that contains key `form.title` with value `$t(shared.entities:organization.title)`
`warehouse` that contains key `form.title` with value `$t(shared.entities:warehouse.title)`

and so on...

the reference key is not included in weblate components as it is shared in an API unique for each deployment. hence `shared.entities` are not a component inside a weblate project.

The outcome after using AutoTranslation is randomly assigning another references as a translation value from one key reference, for example.

`menu.products` is translated as `$t(shared.entities:organization.title)` which should be `$t(shared.entities:product.title)` (some translations are correct while most are like that). Although when you go the the Automatic Suggestion section the translation is correct and preserves the reference.

Knowing that initially the destination languages are initialized with empty string, Also tried removing the language from the UI and resetting them which yielded the same results.

I tired:
- disabling the Translation Memory from the entire weblate application with database reset
- disabling "Auto Translation Propagation"
- disabling "Suggestions"
- disabling "Manage Strings"
- enabling "Enforced checks: [i18next iterpolation]"

foreach component in my project I'm using the following Addons:
![image](https://user-images.githubusercontent.com/122785904/214584378-3fe0bfcd-1629-4093-9d9e-59d063f897ad.png)

**Comment by nijel** [comment]
What Weblate version do you use?

Are the interpolation references correctly highlighted while translating?

**Reply by alikleit** [reply]
I use `v4.15.1`, reference `keys` are highlighted correctly, but not the value in the form view:
![image](https://user-images.githubusercontent.com/122785904/214611957-cfe5ab1d-d083-4436-a572-e1a03e3b67c2.png)


in the matrix view, yes `en,de,es` but wrong translations (not preserving the interpolation)
![image](https://user-images.githubusercontent.com/122785904/214611711-72123584-3284-479b-96a2-d88057e7f42b.png)

**Reply by nijel** [reply]
It's not hightlighted because the interpolation in translation does not match source.

The machine translations are broken due to wrong caching - it ignores the interpolations, so it happily takes results for other interpolated string. I've created https://github.com/WeblateOrg/weblate/issues/8679 to track this.

**Comment by alikleit** [comment]
@nijel 

Getting back to my post last year, we updated weblate to the latest, and we're seeing a similar behavior across different components. 

"title" is a root key in many components, the Automatic Translation which is only set to "Amazon Translate" but no "Translation Memory" is automatically translating titles to the same Translated word which is in another project:

![image](https://github.com/WeblateOrg/weblate/assets/122785904/19ae7396-d9d6-40e0-84eb-650174d68862)

`"Mitglieder"` is auto translated from source `{ "title" : "Members" }`  in another project `portal` - same issue is happening in that project as well in someplaces. 

I tried deleting all the "Translation Memory" records - it seems like they are not being deleted but showing `0` in to the delete button section but keeps the `total` the same.

I also tried to find a switch off button to the Translation Memory, stumbled into the following and made it `off` : 

![image](https://github.com/WeblateOrg/weblate/assets/122785904/51eb8a97-9aa9-4495-82dd-cf42b7ee7f85)

Any way to stop caching, clear cache properly, or a workaround/solution for that?

There is no much on how to turn it off in the docs, where it is mentioned in another link that it is on by default - see installation config here https://docs.weblate.org/en/weblate-4.18/admin/machine.html#mt-weblate-translation-memory


- Update 1

Translation Memory Records can be deleted from the Django Interface  - did a full reset, and the issue is still persisting, seems like the translation memory is working within the project when after I trigger adding components from "component discovery".  Cannot find a way to tun it off from the project so far.  

One example:    

JSON:
```json
{
   ...
  "address-book": {
    "labels": {
      "user_addresses": "My Addresses",
      "organizations": "$t(shared.entities:organization.title_plural)",
      "shop_organization": "$t(shared.entities:organization.title)"
    },
    "product-section": {
    "labels": {
      "shop_organizations": "$t(shared.entities:productShopOrganization.title_plural)",
     }
    }
  ...
}
```

while in the generate translation memory:
```json
...
  {
    "source": "$t(shared.entities:organization.title_plural)",
    "target": "$t(shared.entities:productShopOrganization.title_plural)",
    "source_language": "en",
    "target_language": "de",
    "origin": "runtime-configuration/ui-components",
    "category": 10000004
  },
...
```

Which probably lead to:

![image](https://github.com/WeblateOrg/weblate/assets/122785904/808af134-294c-4751-bc9a-69174f28fa68)

**Reply by nijel** [reply]
Seems like [translation propagation](https://docs.weblate.org/en/latest/admin/projects.html#allow-translation-propagation) might be causing this, the automatic translation is not doing anything different than regular translating here.

**Reply by alikleit** [reply]
@nijel Thank you for the response, we tried:

Added to `environments`: 

```
DEFAULT_TRANSLATION_PROPAGATION=False
```

In the project it self, it was checked, so unchecked it from settings. 
![image](https://github.com/WeblateOrg/weblate/assets/122785904/1c547a49-e690-4b5c-83c2-2614c50a5c05)


Still Getting: 

![image](https://github.com/WeblateOrg/weblate/assets/122785904/36f3efd4-f2e8-4fa7-bfea-5f6e6c9700bb)

same for interpolation:
![image](https://github.com/WeblateOrg/weblate/assets/122785904/aee5e678-5e73-4ab4-bddf-55306c64c860)


Where it should have been:

and for other strings as well (random translation values)

**Reply by nijel** [reply]
Seems like something is wrong in the automatic translation caching. What Weblate version do you use?

**Reply by alikleit** [reply]
we're on `Weblate 5.5.5`

**Reply by nijel** [reply]
I've reopened https://github.com/WeblateOrg/weblate/issues/11179 to track this, it seems like the same issue (even though I miunderstood it before).

**Reply by alikleit** [reply]
@nijel we saw that #11179  was closed, was the issue related to this post also fixed with that commit?

Thanks and appreciated :)

**Reply by nijel** [reply]
Yes, it should be.
